File: packages/php/core/src/Headers.php
<?php

declare(strict_types=1);

namespace Sendmux\Core;

final class Headers
{
    /**
     * @return array{Idempotency-Key?: string}
     */
    public static function idempotency(?string $key = null): array
    {
        return $key === null || $key === '' ? [] : ['Idempotency-Key' => $key];
    }

    /**
     * @return array{If-Match?: string, If-None-Match?: string}
     */
    public static function conditional(?string $ifMatch = null, ?string $ifNoneMatch = null): array
    {
        $headers = [];

        if ($ifMatch !== null && $ifMatch !== '') {
            $headers['If-Match'] = $ifMatch;
        }

        if ($ifNoneMatch !== null && $ifNoneMatch !== '') {
            $headers['If-None-Match'] = $ifNoneMatch;
        }

        return $headers;
    }
}

File: packages/ruby/core/lib/sendmux/core/headers.rb
# frozen_string_literal: true

module Sendmux
  module Core
    module Headers
      def self.idempotency_key(value)
        { idempotency_key: value }
      end

      def self.if_match(value)
        { if_match: value }
      end

      def self.if_none_match(value)
        { if_none_match: value }
      end

      def self.conditional(if_match: nil, if_none_match: nil)
        {}.tap do |headers|
          headers[:if_match] = if_match if if_match
          headers[:if_none_match] = if_none_match if if_none_match
        end
      end
    end
  end
end
